string recognizer
---
string[] object_type_list
